pkg/controller/useraccount/useraccount_util_test.go

```diff
@@ -0,0 +1,110 @@
+package useraccount
+
+import (
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCompareNSTemplateSet(t *testing.T) {
+	tables := []struct {
+		name   string
+		first  toolchainv1alpha1.NSTemplateSetSpec
+		second toolchainv1alpha1.NSTemplateSetSpec
+		want   bool
+	}{
+		{
+			name: "both_same",
+			first: toolchainv1alpha1.NSTemplateSetSpec{
+				TierName: "basic",
+				Namespaces: []toolchainv1alpha1.Namespace{
+					{Type: "cicd", Revision: "rev1", Template: ""},
+					{Type: "ide", Revision: "rev1", Template: ""},
+				},
+			},
+			second: toolchainv1alpha1.NSTemplateSetSpec{
+				TierName: "basic",
+				Namespaces: []toolchainv1alpha1.Namespace{
+					{Type: "ide", Revision: "rev1", Template: ""},
+					{Type: "cicd", Revision: "rev1", Template: ""},
+				},
```

Comment:
Could you please also add a case when the namespaces are in the same order and expect that the `reflect.DeepEqual` should pass as well? For the cases, we would add an additional parameter to the `toolchainv1alpha1.Namespace` object, but forget to change this comparison logic.

Reply:
Hope this will address your concern in other way ;-) check https://github.com/codeready-toolchain/member-operator/pull/53/commits/985cd5b110f0dd3c157fcf6b15e335e601f9ca75

Reply:
great - good idea :+1:  
But still, that doesn't cover the case when we add a new field the NsTemplateSet.spec ...